language/gpt-oss/backends/sglang_backend.py

```diff
@@ -0,0 +1,408 @@
+                )
+            logger.info("Successfully connected to SGLang server")
+            self.initialized = True
+        except Exception as e:
+            logger.error(f"Failed to initialize SGLang backend: {e}")
+            raise
+
+    def _send_request(
+        self,
+        input_ids: List[int],
+        max_tokens: int,
+        temperature: float,
+        top_k: int,
+        top_p: float
+    ) -> Dict[str, Any]:
+        """Send a single request to the SGLang server.
+
+        Args:
+            input_ids: Token IDs for the prompt
+            max_tokens: Maximum tokens to generate
+            temperature: Sampling temperature
+            top_k: Top-k parameter
+            top_p: Top-p parameter
+
+        Returns:
+            Response dictionary from the server
+        """
+        payload = {
+            "input_ids": input_ids,
+            "sampling_params": {
+                "max_new_tokens": max_tokens,
+                "temperature": temperature,
+                "top_k": top_k,
+                "top_p": top_p,
+            }
+        }
+
+        try:
+            response = self.session.post(
+                f"{self.server_url}/generate",
+                json=payload,
+                timeout=self.timeout,
+            )
+            if response.status_code == 200:
+                return response.json()
+            else:
+                logger.error(
+                    f"Request failed with status {response.status_code}: {response.text}"
+                )
+                return {"error": f"HTTP {response.status_code}: {response.text}"}
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Request failed: {e}")
+            return {"error": str(e)}
+
+    def generate(
+        self,
+        prompts: List[List[int]],
+        max_tokens: int = 100,
+        temperature: float = 0.001,
+        top_k: int = 1,
+        top_p: float = 1.0,
+        **kwargs
+    ) -> List[Dict[str, Any]]:
+        """Generate responses for a batch of prompts.
+
+        Args:
+            prompts: List of token ID sequences
+            max_tokens: Maximum tokens to generate per prompt
+            temperature: Sampling temperature
+            top_k: Top-k sampling parameter
+            top_p: Top-p (nucleus) sampling parameter
+            **kwargs: Additional parameters (ignored)
+
+        Returns:
+            List of response dictionaries with keys:
+                - output_ids: List of generated token IDs
+                - output_text: Generated text (if available)
+                - metadata: Additional metadata (latencies, etc.)
+        """
+        if not self.initialized:
+            raise RuntimeError(
+                "Backend not initialized. Call initialize() first.")
+
+        results = []
+        for prompt_ids in prompts:
+            start_time = time.time()
+            response = self._send_request(
```

Comment:
IIUC, this function is a BS=1 singlestream function which we use offline, and the generate_stream is used for server? Wonder if it will be too slow for offline